api/app/tests/hfi/test_hfi_endpoint_request.py

```diff
@@ -21,27 +21,109 @@ def _setup_mock(monkeypatch: pytest.MonkeyPatch):
+
+    def mock_get_all_stations(__):
+        """ Returns mocked WFWXWeatherStations codes. """
+        return all_station_codes
 
-    # mock out database calls:
-    mock_station_crud(monkeypatch)
+    def mock_get_fire_centre_stations(_, __: int):
+        """ Returns mocked WFWXWeatherStation with fuel types. """
+        def get_fuel_type_by_station_code(code: int):
+            if code == code1:
+                return fuel_type_3
+            return fuel_type_2
+        result = []
+        for station_code, planning_area_id in [(code1, 1), (code2, 1), (code1, 2)]:
+            planning_station = PlanningWeatherStation(
+                station_code=station_code, planning_area_id=planning_area_id)
+            fuel_type = get_fuel_type_by_station_code(station_code)
+            result.append((planning_station, fuel_type))
+        return result
+
+    def mock_get_fire_centre_fire_start_ranges(_, __: int):
+        """ Returns mocked FireStartRange """
+        data = ((1, '0-1'), (2, '1-2'), (3, '2-3'), (4, '3-6'), (5, '6+'))
+        return [FireStartRange(id=id, label=range) for id, range in data]
+
+    def mock_get_fire_start_lookup(_):
+        """ Returns mocked FireStartLookup """
+        data = ((1, 1, 1, 1),
+                (2, 1, 2, 1),
+                (3, 1, 3, 2),
+                (4, 1, 4, 3),
+                (5, 1, 5, 4),
+                (6, 2, 1, 1),
+                (7, 2, 2, 1),
+                (8, 2, 3, 2),
+                (9, 2, 4, 4),
+                (10, 2, 5, 5),
+                (11, 3, 1, 2),
+                (12, 3, 2, 3),
+                (13, 3, 3, 4),
+                (14, 3, 4, 5),
+                (15, 3, 5, 6),
+                (16, 4, 1, 3),
+                (17, 4, 2, 4),
+                (18, 4, 3, 4),
+                (19, 4, 4, 5),
+                (20, 4, 5, 6),
+                (21, 5, 1, 4),
+                (22, 5, 2, 5),
+                (23, 5, 3, 6),
+                (24, 5, 4, 6),
+                (25, 5, 5, 6))
+        return [FireStartLookup(id=id,
+                                fire_start_range_id=fire_start_range_id,
+                                mean_intensity_group=mean_intensity_group,
+                                prep_level=prep_level) for
+                id, fire_start_range_id, mean_intensity_group, prep_level in data]
+
+    fuel_types = [
+        fuel_type_1,
+        fuel_type_2,
+        fuel_type_3
+    ]
+
+    def mock_get_fuel_type_by_id(_, fuel_type_id: int):
+        """ Returns mocked FuelType """
+        return next(fuel_type for fuel_type in fuel_types if fuel_type.id == fuel_type_id)
+
+    def mock_get_fuel_types(_):
+        return fuel_types
+
+    monkeypatch.setattr(app.hfi.hfi_calc, 'get_fire_weather_stations', mock_get_fire_weather_stations)
+    monkeypatch.setattr(app.db.crud.hfi_calc, 'get_all_stations', mock_get_all_stations)
+    # TODO: this is problematic, why are we calling get_fire_centre_stations twice?
```

Comment:
```suggestion
```

Reply:
shouldn't need this comment in the test file.

Reply:
I'd like to keep it there - it's being mocked out twice 
`monkeypatch.setattr(app.hfi.hfi_calc, 'get_fire_centre_stations', mock_get_fire_centre_stations)`
and
`monkeypatch.setattr(app.routers.hfi_calc, 'get_fire_centre_stations', mock_get_fire_centre_stations)`
It needs to be addressed, but this PR is just too busy already.